src/main/java/macrobase/analysis/classify/BatchingPercentileClassifier.java

```diff
@@ -0,0 +1,57 @@
+package macrobase.analysis.classify;
+
+import com.google.common.collect.Lists;
+import macrobase.analysis.result.DatumWithNorm;
+import macrobase.analysis.result.OutlierClassificationResult;
+import macrobase.conf.MacroBaseConf;
+import macrobase.conf.MacroBaseDefaults;
+import macrobase.datamodel.Datum;
+import macrobase.util.IterUtils;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class BatchingPercentileClassifier extends OutlierClassifier {
+    Iterator<OutlierClassificationResult> resultIterator;
+
+    final double targetPercentile;
+
+    public BatchingPercentileClassifier(MacroBaseConf conf, Iterator<Datum> input) {
+        this(conf, input, conf.getDouble(MacroBaseConf.TARGET_PERCENTILE,
+                                         MacroBaseDefaults.TARGET_PERCENTILE));
+    }
+
+    public BatchingPercentileClassifier(MacroBaseConf conf, Iterator<Datum> input, double percentile) {
+        super(conf, input);
+        this.targetPercentile = percentile;
+    }
+
+    @Override
+    public OutlierClassificationResult next() {
+        if(resultIterator == null) {
```

Comment:
@deepakn94 This is a convenience wrapper for the existing batch functionality. Is your proposal that we should check the input for more tuples to drain?